### PR TITLE
Added the address_type_controls to BGP Route Summarization resource

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added `summary-only` to the `ctrl` attribute.
Also added the `address_type_controls` to BGP Route Summarization resource

### CODE REVIEW

File: website/docs/r/bgp_route_summarization.html.markdown [hunk 1]
```diff
@@ -11,32 +11,38 @@ description: |-
...
-- `ctrl` - (Optional) The control state.
-  Allowed values: "as-set", "none". Default value: "none".
-- `name_alias` - (Optional) Name alias for object BGP route summarization.
+- `tenant_dn` - (Required) Distinguished name of the parent Tenant object. Type: String.
+- `name` - (Required) Name of the BGP Route Summarization object. Type: String.
+- `annotation` - (Optional) Annotation of the BGP Route Summarization object. Type: String.
+- `description` - (Optional) Description of the BGP Route Summarization object. Type: String.
+- `attrmap` - (Optional) Route Map Summary of the BGP Route Summarization object. Type: String.
+- `ctrl` - (Optional) Control State of the BGP Route Summarization object. Allowed values are "as-set", "summary-only". Type: List.
+- `address_type_controls` - (Optional) Address Type Controls of the BGP Route Summarization object. Allowed values are "af-ucast", "af-mcast", "af-label-ucast". Default is "af-ucast". Type: List.
```
Comment (akinross) [inline website/docs/r/bgp_route_summarization.html.markdown:44]: Is this default correct?

**Review by akinross — DISMISSED** [review]
LGTM

File: website/docs/r/bgp_route_summarization.html.markdown [hunk 2]
```diff
@@ -11,32 +11,38 @@ description: |-
...
 
+## Example Usage
+
+```hcl
+resource "aci_bgp_route_summarization" "bgp_rt_summ_pol" {
+  tenant_dn             = aci_tenant.tf_tenant.id
+  name                  = "bgp_rt_summ_pol"
+  description           = "from terraform"
+  attrmap               = "sample attrmap"
+  ctrl                  = sort(["summary-only", "as-set"])                 # Use the sorted list to handle identical changes
```
Comment (shrsr) [inline website/docs/r/bgp_route_summarization.html.markdown:31]: Remove comments # Use the sorted list to handle identical changes and # Use the sorted list to handle identical changes

File: examples/bgp_route_summarization/main.tf [hunk 3]
```diff
@@ -14,19 +14,15 @@ provider "aci" {
...
-  attrmap    = "example"
-  ctrl       = "as-set"
-  name_alias = "example"
-
+resource "aci_bgp_route_summarization" "bgp_rt_summ_pol" {
+  tenant_dn             = aci_tenant.tf_tenant.id
+  name                  = "bgp_rt_summ_pol"
+  description           = "from terraform"
+  attrmap               = "sample attrmap"
+  ctrl                  = sort(["summary-only", "as-set"])                 # Use the sorted list to handle identical changes
```
Comment (shrsr) [inline examples/bgp_route_summarization/main.tf:26]: What does this mean ->Use the sorted list to handle identical changes?

Isn't TF able to differentiate between  ["summary-only", "as-set"]  and ["as-set", "summary-only"]?

**Review by shrsr — CHANGES_REQUESTED** [review]

**Review by shrsr — CHANGES_REQUESTED** [review]

**Review by akinross — DISMISSED** [review]
LGTM

File: aci/utils.go [hunk 4]
```diff
@@ -248,6 +248,35 @@ func suppressBitMaskDiffFunc() func(k, old, new string, d *schema.ResourceData)
 	}
 }
 
+// suppressTypeListDiffFunc returns the bool value (true) when the old and new list is identical for a given key.
+// If the old and new list not is identical for a given key then it returns false.
+// The old and new list should be retrieved from the state and configuration file using GetChange.
+func suppressTypeListDiffFunc(key, oldValue, newValue string, d *schema.ResourceData) bool {
```
[the remark this answers is not in the document]
Comment (sajagana) [reply aci/utils.go:254]: Created an issue to track other resource changes: https://github.com/CiscoDevNet/terraform-provider-aci/issues/1044

**Review by shrsr — DISMISSED** [review]
LGTM (Discussed the comments with Sabari )

**Review by anvitha-jain — DISMISSED** [review]
LGTM

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_bgprtsummpol.go [hunk 5]
```diff
@@ -23,44 +25,103 @@ func resourceAciBgpRouteSummarization() *schema.Resource {
...
+				Computed: true,
+			},
 			"ctrl": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
 				Computed: true,
 				ValidateFunc: validation.StringInSlice([]string{
-					"as-set", "none",
+					"as-set",
+					"summary-only",
```
[the remark this answers is not in the document]
Comment (sajagana) [reply aci/resource_aci_bgprtsummpol.go:108]: The ctrl attribute allows to set empty(none) string, but none is not in the list of valid values.

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**